runelite-api/src/main/java/net/runelite/api/geometry/Geometry.java

```diff
@@ -0,0 +1,464 @@
+				start[1] = coords[1];
+				prevCoords[0] = coords[0];
+				prevCoords[1] = coords[1];
+			}
+			else if (type == PathIterator.SEG_LINETO)
+			{
+				Point2D.Float intersection = lineIntersectionPoint(
+					prevCoords[0], prevCoords[1], coords[0], coords[1], x1, y1, x2, y2);
+				if (intersection != null)
+				{
+					intersections.add(intersection);
+				}
+				prevCoords[0] = coords[0];
+				prevCoords[1] = coords[1];
+			}
+			else if (type == PathIterator.SEG_CLOSE)
+			{
+				Point2D.Float intersection = lineIntersectionPoint(
+					coords[0], coords[1], start[0], start[1], x1, y1, x2, y2);
+				if (intersection != null)
+				{
+					intersections.add(intersection);
+				}
+			}
+			it.next();
+		}
+
+		return intersections;
+	}
+
+	/**
+	 * Transforms the points in a path according to a method.
+	 *
+	 * @param it The iterator of the path to change the points on.
+	 * @param method The method to use to transform the points. Takes a float[2] array with x and y coordinates as parameter.
+	 * @return The transformed path.
+	 */
+	public static GeneralPath transformPath(PathIterator it, Consumer<float[]> method)
+	{
+		GeneralPath path = new GeneralPath();
+		float[] coords = new float[2];
+		while (!it.isDone())
+		{
+			int type = it.currentSegment(coords);
+			if (type == PathIterator.SEG_MOVETO)
+			{
+				method.accept(coords);
+				path.moveTo(coords[0], coords[1]);
+			}
+			else if (type == PathIterator.SEG_LINETO)
+			{
+				method.accept(coords);
+				path.lineTo(coords[0], coords[1]);
+			}
+			else if (type == PathIterator.SEG_CLOSE)
+			{
+				path.closePath();
+			}
+			it.next();
+		}
+
+		return path;
+	}
+
+	/**
+	 * Transforms the points in a path according to a method.
+	 *
+	 * @param path The path to change the points on.
+	 * @param method The method to use to transform the points. Takes a float[2] array with x and y coordinates as parameter.
+	 * @return The transformed path.
+	 */
+	public static GeneralPath transformPath(GeneralPath path, Consumer<float[]> method)
+	{
+		return transformPath(path.getPathIterator(new AffineTransform()), method);
+	}
+
+	/**
+	 * Splits a line into smaller segments and appends the segments to a path.
+	 *
+	 * @param path The path to append lines to.
+	 * @param unitSize The desired length to use for the segmented lines.
+	 * @param x1 X coordinate of the first endpoint of the line.
+	 * @param y1 Y coordinate of the first endpoint of the line.
+	 * @param x2 X coordinate of the second endpoint of the line.
+	 * @param y2 Y coordinate of the second endpoint of the line.
+	 */
+	private static void appendUnitLines(GeneralPath path, float unitSize,
+		float x1, float y1, float x2, float y2)
+	{
+		float x = x1;
+		float y = y1;
+		float angle = (float)Math.atan2(y2 - y1, x2 - x1);
+		float dx = (float)Math.cos(angle) * unitSize;
+		float dy = (float)Math.sin(angle) * unitSize;
+		float length = (float)Math.hypot(x2 - x1, y2 - y1);
+		int steps = (int)((length - 1e-4) / unitSize);
```

Comment:
It's more like a Math.floor. If we round up the lines could go past the end. The int cast is there because if `steps` is something like 4.5, we'd go too far as we'd take 5 steps.
The idea behind the magic number is to always round down the result to remove negligible lines with length of something like 0.000000001 due to precision errors of floating-point values.